packages/server/src/main/java/software/uncharted/terarium/hmiserver/utils/rebac/ReBACService.java

```diff
@@ -77,12 +84,51 @@ public class ReBACService {
 
 	public static String API_SERVICE_USER_ID;
 
-	volatile String CURRENT_ZED_TOKEN;
+	private volatile String CURRENT_ZED_TOKEN;
 
 	private String getKeycloakBearerToken() {
 		return "Bearer " + keycloak.tokenManager().getAccessTokenString();
 	}
 
+	private class CacheKey {
+
+		SchemaObject who;
+		Schema.Permission permission;
+		SchemaObject what;
+
+		CacheKey(SchemaObject who, Schema.Permission permission, SchemaObject what) {
+			this.who = who;
+			this.permission = permission;
+			this.what = what;
+		}
```

Comment:
Is this the contructor of the class?

Reply:
Yes that is a constructor (the only one for this class)